pkg/docker/config/config.go

```diff
@@ -599,13 +614,27 @@ func findAuthentication(registry, path string, legacyFormat bool) (types.DockerA
 		normalizedAuths[normalizeRegistry(k)] = v
 	}
 
-	if val, exists := normalizedAuths[registry]; exists {
+	// If a reference path is provided, then we include it in the auth registry
+	// to be looked up.
+	registryLookup := registry
+	if refPath != "" {
+		registryLookup = stripImageName(registry, refPath)
+	}
```

Comment:
We always pass a `refPath` around. So if we have a pre-existing auth.json with a credentials for `quay.io`, would we still be able to find them when pulling `quay.io/repo/image:tag`?

Doesn't that break backwards compat as we'd always look for credentials for `quay.io/repo/image`?

Reply:
Yes, with the latest implementation.